.github/workflows/ci-e2e-test.yaml

```diff
@@ -1,15 +1,120 @@
+      - name: "E2E Detection"
+        id: e2es
+        run: |
+          scenarios=$(find ${PWD}/e2e -type f -name 'e2e_test.go' -exec dirname {} \; | xargs -n 1 basename | jq -R -s -c 'split("\n")[:-1]')
+          echo $scenarios
+          echo "result=$scenarios" >> $GITHUB_OUTPUT
+
   e2e:
-    runs-on: "default"
+    name: "Run ${{ matrix.e2es }}(${{ matrix.k8s-version}}) E2E"
+    if: needs.init.outputs.tests != '[]'
+    needs: [ init ]
+    runs-on: [ default ]
+    env:
+      ADMIN_CLUSTER: greenhouse-admin
+      REMOTE_CLUSTER: greenhouse-remote
+
     strategy:
       fail-fast: false
       matrix:
-        k8s-version: ["v1.29.9", "v1.30.5", "v1.31.1"]
+        k8s-version: [ "v1.29.8", "v1.30.4", "v1.31.0" ]
+        e2es: ${{fromJson(needs.init.outputs.tests)}}
     steps:
-      - uses: cloudoperators/common/workflows/e2e@main
+      - name: "Checkout"
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      # Create the admin cluster with latest kubernetes version
+      - name: Create Admin Cluster
+        uses: helm/kind-action@v1.10.0
         with:
-          kubernetes-version: ${{ matrix.k8s-version }}
+          version: 'v0.24.0'
+          cluster_name: ${{ env.ADMIN_CLUSTER }}
+
+      # Create the remote cluster with kubernetes version from the matrix
+      - name: Create Remote Cluster
+        uses: helm/kind-action@v1.10.0
+        with:
+          version: 'v0.24.0'
+          node_image: 'kindest/node:${{ matrix.k8s-version }}'
+          cluster_name: ${{ env.REMOTE_CLUSTER }}
+          config: ${{ github.workspace }}/e2e/kind-config.yaml
+
+      # build CLI, setup e2e environment and prepare kubeconfigs
+      - name: "Prepare E2E Config"
+        id: config
+        run: |
+          make setup-e2e
+          echo "admin_config=$GITHUB_WORKSPACE/bin/${{env.ADMIN_CLUSTER}}.kubeconfig" >> $GITHUB_OUTPUT
+          echo "remote_config=$GITHUB_WORKSPACE/bin/${{env.REMOTE_CLUSTER}}.kubeconfig" >> $GITHUB_OUTPUT
+          echo "remote_int_config=$GITHUB_WORKSPACE/bin/${{env.REMOTE_CLUSTER}}-int.kubeconfig" >> $GITHUB_OUTPUT
+
+      # run the e2e tests with the scenario from the matrix
+      - name: "E2E Run"
+        id: e2e
+        continue-on-error: true
+        env:
+          SCENARIO: ${{ matrix.e2es }}
+          EXECUTION_ENV: gh-actions
+          GREENHOUSE_ADMIN_KUBECONFIG: ${{ steps.config.outputs.admin_config }}
+          GREENHOUSE_REMOTE_KUBECONFIG: ${{ steps.config.outputs.remote_config }}
+          GREENHOUSE_REMOTE_INT_KUBECONFIG: ${{ steps.config.outputs.remote_int_config }}
+          CONTROLLER_LOGS_PATH: ${{github.workspace}}/bin/${{matrix.e2es}}-${{matrix.k8s-version}}.txt
+        run: |
+          go install github.com/onsi/ginkgo/v2/ginkgo@latest
+          ginkgo version
```

Comment:
Do we need this, or is it only a leftover?

Reply:
Left over will remove it.